# origin: Debian (herbert)
# cset: n/a
# inclusion: not submitted
# description: do not get seqno if we will not use it
# revision date: 2004-09-04

diff -urN kernel-source-2.4.26/fs/isofs/inode.c kernel-source-2.4.26-1/fs/isofs/inode.c
--- kernel-source-2.4.26/fs/isofs/inode.c	2002-11-29 10:53:15.000000000 +1100
+++ kernel-source-2.4.26-1/fs/isofs/inode.c	2002-11-30 15:49:27.000000000 +1100
@@ -1148,7 +1148,10 @@
 	struct iso_directory_record * tmpde = NULL;
 	unsigned int de_len;
 	unsigned long offset;
-	int volume_seq_no, i;
+	int i;
+#ifndef IGNORE_WRONG_MULTI_VOLUME_SPECS
+	int volume_seq_no;
+#endif
 
 	bh = sb_bread(inode->i_sb, block);
 	if (!bh)
@@ -1280,9 +1275,6 @@
 		test_and_set_gid(&inode->i_gid, inode->i_sb->u.isofs_sb.s_gid);
 	}
 
-	/* get the volume sequence number */
-	volume_seq_no = isonum_723 (de->volume_sequence_number) ;
-
     /*
      * Multi volume means tagging a group of CDs with info in their headers.
      * All CDs of a group must share the same vol set name and vol set size
@@ -1292,6 +1284,9 @@
      * of ftp://ftp.ecma.ch/ecma-st/Ecma-119.pdf (ECMA 119 2nd Ed = ISO 9660)
      */
 #ifndef IGNORE_WRONG_MULTI_VOLUME_SPECS
+	/* get the volume sequence number */
+	volume_seq_no = isonum_723 (de->volume_sequence_number) ;
+
 	/*
 	 * Disable checking if we see any volume number other than 0 or 1.
 	 * We could use the cruft option, but that has multiple purposes, one

